### PR TITLE
ICDS Dashboard: port basic version of block view to mobile

### PR DESCRIPTION
**Pull request by czue**
image

this was relatively easy, though there are a bunch of mechanical test changes

### CODE REVIEW

File: custom/icds_reports/static/icds_reports/js/spec/enrolled-children.directive.spec.js [hunk 1]
```diff
@@ -14,24 +14,7 @@ describe('Enrolled Children Directive', function () {
...
-            {id: '24', name: '12-24 months'},
-            {id: '36', name: '24-36 months'},
-            {id: '48', name: '36-48 months'},
-            {id: '60', name: '48-60 months'},
-            {id: '72', name: '60-72 months'},
-        ]);
-        $provide.constant("userLocationId", null);
-        $provide.constant("haveAccessToAllLocations", false);
-        $provide.constant("isAlertActive", false);
+        utils.provideDefaultConstants($provide, true, true);
```
Comment (orangejenny) [inline custom/icds_reports/static/icds_reports/js/spec/enrolled-children.directive.spec.js]: Not blocking: I wish js supported named params; the calls with multiple booleans are hard to read and easy to mess up. What would you think of making `provideDefaultConstants` consume a parameters object instead? Like:
```
module.provideDefaultConstants = function ($provide, options) {
   hqImport("hqwebapp/js/assert_properties").assert(options, [], ['includeGenders', 'includeAges', 'overrides']);
   ...
   if (options.includeGenders) {
      ...
```

**Review by orangejenny — APPROVED** [review]

File: custom/icds_reports/static/icds_reports/js/spec/enrolled-children.directive.spec.js (hunk 1 above)
Comment (czue) [reply custom/icds_reports/static/icds_reports/js/spec/enrolled-children.directive.spec.js]: ha: https://github.com/dimagi/commcare-hq/pull/26140#discussion_r355191976

I agree that would be better. I got hung up with providing both `options` and `overrides` though I guess I could make `overrides` a property of options. 

I'll make a backlog item for this and hopefully get to it soonish.

Comment (orangejenny) [reply custom/icds_reports/static/icds_reports/js/spec/enrolled-children.directive.spec.js]: Haha, glad we're aligned. Doing `overrides` as a separate param and just using options for the booleans seems fine, too.

Comment (czue) [reply custom/icds_reports/static/icds_reports/js/spec/enrolled-children.directive.spec.js]: https://github.com/dimagi/commcare-hq/pull/26484

**Pull request merged**